### PR TITLE
EVG-19909 Run cmake-test on macOS but skip flaky tests

### PR DESCRIPTION
**Pull request by lgfang**
**Jira Ticket:** [EVG-19909](https://jira.mongodb.org/browse/EVG-19909)  

**Whats Changed:**  

-   Split the test case "Actual Actor Example" to "Actual Actor Example" and  
    "Actual Actor Example Timing"
-   Modify 'run-genny cmake-test' to accept the option '&#x2013;regex-exclude'
-   Modify 'evergreen.yml' run cmake-test on macOS with '&#x2013;regex-exclude'

At the moment, the `f_cmake_test_ignore_flaky` function runs cmake-test with  
`--regex-exclude` to skip timing/random number related tests which are known to  
fail on macOS. In the future, should more such tests are discovered, please skip  
them similarly.  

**Patch testing results:**  
<https://spruce.mongodb.com/version/6492ab750ae6065cd41edb29/tasks>  
<https://spruce.mongodb.com/version/649a474457e85a45efc31d70/tasks>  
<https://spruce.mongodb.com/version/649a46c0d1fe074124687ff2/tasks>

### CODE REVIEW

**Review by thessem — APPROVED** [review]
Happy with this approach. In terms of API I like `regex-exclude` and the fact it doesn't scale well or have a lot of extra functionality is an indicator that "we shouldn't let the list get too long here". It passing on essentially the same cmake test parameter is nice.

I think this achieves the result of making an essentially "passing" build pass without lying to the user. MacOS is not a good platform for performance testing on _because_ those tests fail, we shouldn't hide it, but for every day development it doesn't matter.

**Comment by PyGeek03** [comment]
1. From EVG-19909, a flaky test investigated is "value_generators_test:YAML". It seems like the regex doesn't have this test?
2. Have we investigated why we have to skip the "Actual Actor Example" tests?

**Review by PyGeek03 — CHANGES_REQUESTED** [review]
1. From [EVG-19776](https://jira.mongodb.org/browse/EVG-19776), a flaky test investigated is "value_generators_test:YAML". It seems like the regex doesn't have this test?
2. Have we investigated why we have to skip the "Actual Actor Example" tests?

**Comment by lgfang** [comment]
>  From [EVG-19776](https://jira.mongodb.org/browse/EVG-19776), a flaky test investigated is "value_generators_test:YAML". It seems like the regex doesn't have this test?

According to my test, this one never fails when run by evergreen.

> Have we investigated why we have to skip the "Actual Actor Example" test, and whether we should do that? If the issue is with a specific subtest within this test, should we split that subtest out to be its own test and ignore only that test?

Yes, this one fails due to timing similar to the sleepNonBlocking .

Given that the goal of this ticket is very if the "essential" build tests succeed, I think it is ok to skip such tests.

**Review by PyGeek03 — APPROVED** [review]
> > From [EVG-19776](https://jira.mongodb.org/browse/EVG-19776), a flaky test investigated is "value_generators_test:YAML". It seems like the regex doesn't have this test?
> 
> According to my test, this one never fails when run by evergreen.

It's interesting how our own Mac ARM laptops are also different from Evergreen's. But yeah I agree with you on this. Let's keep the test on Evergreen as a documentation of this fact.
 
> > Have we investigated why we have to skip the "Actual Actor Example" test, and whether we should do that? If the issue is with a specific subtest within this test, should we split that subtest out to be its own test and ignore only that test?
> 
> Yes, this one fails due to timing similar to the sleepNonBlocking .
> 
> Given that the goal of this ticket is very if the "essential" build tests succeed, I think it is ok to skip such tests.

I don't know if the "Actual Actor Example" test is essential or not, but I trust your judgement on this one.

Thank you for this PR, I really like your solution of using the regex-exclude feature in ctest to implement this!

**Review by PyGeek03 — APPROVED** [review]
This looks really awesome, thank you Lungang!

**Comment by lgfang** [comment]
Thanks for Minh's suggestion, I've split the "Actual Actor Example" to "Actual Actor Example" and "Actual Actor Example Timing" and ignore the later only.

File: evergreen.yml [hunk 1]
```diff
@@ -349,6 +364,21 @@ functions:
...
+  ##
+  f_cmake_test_ignore_flaky:
+  - command: subprocess.exec
+    params:
+      binary: ./src/genny/run-genny
+      args:
+      - -v
+      - cmake-test
+      - --regex-exclude
+      - "(sleepNonBlocking|Non-Blocking end|Actual Actor Example Timing)"
```
Comment (rtimmons) [inline evergreen.yml:380]: Consider a standard test name e.g. IgnoredOnMacOs so the tests can be marked ignored more explicitly and we don't accidentally break CI by renaming tests.

Comment (lgfang) [reply evergreen.yml:380]: Just to confirm I understand you correctly, you mean renaming "Acutual Actor Example Timing" to "Actual Actor Example IgnoredOnMacOS"?

Comment (rtimmons) [reply evergreen.yml:380]: Yes. That puts the test together with the specification to skip on macos. Again, just a suggestion.

Comment (lgfang) [reply evergreen.yml:380]: Good idea. It will also make the regex simpler.
Since I've already merged this PR, I'm creating a PR for this renaming.

**Pull request merged**